Detectors/DCS/include/DetectorsDCS/DCSProcessor.h

```diff
@@ -0,0 +1,122 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef DETECTOR_DCS_DCSPROCESSOR_H_
+#define DETECTOR_DCS_DCSPROCESSOR_H_
+
+#include <memory>
+#include <Rtypes.h>
+#include <unordered_map>
+#include <deque>
+#include "Framework/Logger.h"
+#include "DetectorsDCS/DataPointCompositeObject.h"
+#include "DetectorsDCS/DataPointIdentifier.h"
+#include "DetectorsDCS/DataPointValue.h"
+#include "DetectorsDCS/DeliveryType.h"
+
+/// @brief Class to process DCS data points
+
+namespace o2
+{
+namespace dcs
+{
+
+class DCSProcessor
+{
+
+  using Ints = std::vector<int>;
+  using Chars = std::vector<char>;
+  using Doubles = std::vector<double>;
+  using Binaries = std::array<uint64_t, 7>;
+  using Strings = std::array<char, 56>;
+
+  using DQChars = std::deque<char>;
+  using DQInts = std::deque<int>;
+  using DQDoubles = std::deque<double>;
+  using DQUInts = std::deque<uint32_t>;
+  using DQBools = std::deque<bool>;
+  using DQStrings = std::deque<Strings>;
+  using DQTimes = std::deque<uint32_t>;
+  using DQBinaries = std::deque<Binaries>;
+
+  using DPID = o2::dcs::DataPointIdentifier;
+  using DPVAL = o2::dcs::DataPointValue;
+  using DPCOM = o2::dcs::DataPointCompositeObject;
+
+ public:
+  DCSProcessor() = default;
+  ~DCSProcessor() = default;
+
+  void init(std::vector<DPID> aliaseschars, std::vector<DPID> aliasesints,
+            std::vector<DPID> aliasesdoubles, std::vector<DPID> aliasesUints,
+            std::vector<DPID> aliasesbools, std::vector<DPID> aliasesstrings,
+            std::vector<DPID> aliasestimes, std::vector<DPID> aliasesbinaries);
+
+  void init(std::vector<DPID> aliases);
+
+  int process(const std::unordered_map<DPID, DPVAL>& map);
+
+  void processAlias(DPID& alias, DeliveryType type, const std::unordered_map<DPID, DPVAL>& map, int& count, std::unordered_map<DPID, DPVAL>::const_iterator& it);
+
+  void processChars();
+  void processInts();
+  void processDoubles();
+  void processUInts();
+  void processBools();
+  void processStrings();
+  void processTimes();
+  void processBinaries();
+  uint64_t processFlag(uint64_t flag, const char* alias);
+
+  void doSimpleMovingAverage(int nelements, std::deque<int>& vect, float& avg, bool& isSMA);
```

Comment:
should not this be templated wrt ``deque``?

Reply:
Yes, this I can indeed template.

Reply:
Done